test/runner/plugins/deferred.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const kDeferred = Symbol('deferred');
+
+(mocha => {
+  const Context = mocha.Context;
+  function makeExecuteDeferred(test) {
+    return () => {
+      const deferredActions = test[kDeferred];
+
+      return Promise.all(
+        Array.from(deferredActions).map(action => {
+          if (action.length > 0) {
+            // assume these are async methods with provided `done`
+            return new Promise((resolve, reject) => {
+              function done(err) {
+                if (err) return reject(err);
+                resolve();
+              }
+
+              action(done);
+            });
+          }
+
+          // otherwise assume a Promise is returned
+          action();
```

Comment:
i think this needs to be `return action();` to return the `Promise`?

Reply:
thanks, nice catch